server/socketServer.ts

```diff
@@ -26,18 +26,27 @@ const io = new Server(socketConfig);
 // and remembers the original insertion order of the keys
 // const socketToPlayerId = new Map<number, string>();
 const playerData = new Map<string, PlayerData>();
-// const playerData: [Player] = [];
+const playerClients: string[] = [];
 
 io.on('connection', (socket: SocketData) => {
-  socket.broadcast.emit('playerConnect', socket.id);
+  playerClients.push(socket.id);
+  socket.emit('playerAssignment', socket.id);
+
+  console.log(playerClients);
+  io.emit('playerConnect', playerClients);
```

Comment:
All connected clients need to know when a new client joins so they can all have the correct client list
`io.emit()` will broadcast the message to all connected clients, not just the sender
https://socket.io/docs/v4/emit-cheatsheet/